### PR TITLE
Enable allocating a context with vendor data

### PR DESCRIPTION
**Pull request by yishaih**
Extend verbs_device_ops to get vendor data as part of alloc_context, this functionality is used by mlx5 driver to allocate a DEVX context.

### CODE REVIEW

File: libibverbs/libibverbs.map.in [hunk 1]
```diff
@@ -178,6 +178,7 @@ IBVERBS_PRIVATE_@IBVERBS_PABI_VERSION@ {
 		ibv_query_gid_type;
 		ibv_register_driver;
 		verbs_allow_disassociate_destroy;
+		verbs_open_device;
```
Comment (jgunthorpe) [inline libibverbs/libibverbs.map.in]: Need to update IBVERBS_PABI_VERSION

File: providers/bnxt_re/main.c [hunk 2]
```diff
@@ -110,13 +110,17 @@ static const struct verbs_context_ops bnxt_re_cntx_ops = {
...
-						   int cmd_fd)
+						   int cmd_fd,
+						   void *vendor_attr)
 {
 	struct ibv_get_context cmd;
 	struct ubnxt_re_cntx_resp resp;
 	struct bnxt_re_dev *dev = to_bnxt_re_dev(vdev);
 	struct bnxt_re_context *cntx;
 
+	if (vendor_attr)
```
Comment (jgunthorpe) [inline providers/bnxt_re/main.c]: Why? If 'vendor_attr' is internal to the driver we don't need to check it everywhere like this.

File: libibverbs/device.c [hunk 3]
```diff
@@ -266,9 +266,7 @@ static void set_lib_ops(struct verbs_context *vctx)
 	vctx->create_cq_ex = __lib_ibv_create_cq_ex;
 }
 
-LATEST_SYMVER_FUNC(ibv_open_device, 1_1, "IBVERBS_1.1",
-		   struct ibv_context *,
-		   struct ibv_device *device)
+struct ibv_context *verbs_open_device(struct ibv_device *device, void *vendor_attr)
```
Comment (jgunthorpe) [inline libibverbs/device.c]: Stop using the word vendor.

Call this private_data

File: libibverbs/libibverbs.map.in (hunk 1 above)
Comment (yishaih) [reply libibverbs/libibverbs.map.in]: OK, PR was updated

**Pull request merged**